mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipperTest.java

```diff
@@ -177,6 +189,66 @@ void archiveShouldWriteMailboxMetadataWhenPresent() throws Exception {
         }
     }
 
+    @Test
+    void archiveShouldWriteMailBoxWithoutAnAnnotationSubDirWhenEmpty() throws Exception {
+        testee.archive(ImmutableList.of(MAILBOX_1_WITHOUT_ANNOTATION), Stream.of(), output);
+
+        try (ZipFile zipFile = new ZipFile(toSeekableByteChannel(output))) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MAILBOX_1.getName() + "/")
+                );
+        }
+    }
+
+    @Test
+    void archiveShouldWriteMailboxAnnotationsInASubDirWhenPresent() throws Exception {
+        testee.archive(ImmutableList.of(new MailboxWithAnnotations(MAILBOX_1, WITH_ANNOTATION_1)), Stream.of(), output);
+
+        try (ZipFile zipFile = new ZipFile(toSeekableByteChannel(output))) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MAILBOX_1.getName() + "/"),
+                    hasName(MAILBOX_1.getName() + "/annotations/").isDirectory(),
+                    hasName(MAILBOX_1.getName() + "/annotations/" + ANNOTATION_1.getKey().asString())
+                );
+        }
+    }
+
+    @Test
+    void archiveShouldWriteMailboxAnnotationsInASubDirWhenTwoPresent() throws Exception {
+        testee.archive(ImmutableList.of(new MailboxWithAnnotations(MAILBOX_1, WITH_ANNOTATION_1_AND_2)), Stream.of(), output);
+
+        try (ZipFile zipFile = new ZipFile(toSeekableByteChannel(output))) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MAILBOX_1.getName() + "/"),
+                    hasName(MAILBOX_1.getName() + "/annotations/").isDirectory(),
+                    hasName(MAILBOX_1.getName() + "/annotations/" + ANNOTATION_1.getKey().asString())
+                        .hasStringContent(ANNOTATION_1_CONTENT),
+                    hasName(MAILBOX_1.getName() + "/annotations/" + ANNOTATION_2.getKey().asString())
+                        .hasStringContent(ANNOTATION_2_CONTENT)
+                );
+        }
+    }
+
+    @Test
+    void archiveShouldWriteMailboxAnnotationsInASubDirWhenTwoPresentWithTheSameName() throws Exception {
+        testee.archive(ImmutableList.of(new MailboxWithAnnotations(MAILBOX_1, ImmutableList.of(ANNOTATION_1, ANNOTATION_1_BIS))), Stream.of(), output);
+
+        try (ZipFile zipFile = new ZipFile(toSeekableByteChannel(output))) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MAILBOX_1.getName() + "/"),
+                    hasName(MAILBOX_1.getName() + "/annotations/").isDirectory(),
```

Comment:
Can you add tests trying to mess things up with a `mailbox/annotations` mailbox?

Reply:
I will create a new ticket to url encode the mailbox name.